pkgs/tools/system/lanzaboote/stub.nix

```diff
@@ -0,0 +1,32 @@
+{ rustPlatform, stdenv, lib, fetchFromGitHub }:
+
+rustPlatform.buildRustPackage
+rec {
+  pname = "lanzaboote_stub";
+  version = "0.3.0";
+  src = fetchFromGitHub {
+    owner = "nix-community";
+    repo = "lanzaboote";
+    rev = "v${version}";
+    hash = "sha256-Fb5TeRTdvUlo/5Yi2d+FC8a6KoRLk2h1VE0/peMhWPs=";
+  };
+
+  sourceRoot = "source/rust/stub";
+
+  cargoLock = {
+    lockFile = ./Cargo-stub.lock;
+  };
```

Comment:
I don't think we're allowed to `readFile` from derivation outputs in nixpkgs, are we? Even if it's a fixed-output-derivation? Because that results in needing to perform a build (fetch the source) just to eval the `buildRustPackage` derivation.

Reply:
I'm more in favor of more lockfiles in nixpkgs, especially for the Rust case, but that's fair.

Reply:
cargo lock files are big and bloat the repo size and building each crate individually cripples remote builder speed and slows things dramatically down.

Reply:
> cargo lock files are big and bloat the repo size and building each crate individually cripples remote builder speed and slows things dramatically down.

That's not the problem of the lockfiles, that's a Nix and Git problem.

Reply:
@amjoseph-nixpkgs given your last reaction, I am now confused if you are against or in favor of this :D. Or if you disagree that Cargo lockfiles slowing down Git is a Git problem and individual crate building slowing down Nix remote builder protocol is a Nix problem?

Reply:
Anyway, I got rid of them. If someone wants them again, they can do a follow up PR and convince everyone.